test/addons/register-signal-handler/binding.cc

```diff
@@ -0,0 +1,33 @@
+#ifndef _WIN32
+#include <node.h>
+#include <v8.h>
+#include <uv.h>
+#include <assert.h>
+#include <unistd.h>
+
+using v8::Boolean;
+using v8::FunctionCallbackInfo;
+using v8::Int32;
+using v8::Value;
+
+void Handler(int signo, siginfo_t* siginfo, void* ucontext) {
+  char signo_char = signo;
+  int written = write(1, &signo_char, 1);  // write() is signal-safe.
```

Comment:
It is however interruptible. You should retry on EINTR.

Reply:
Done!